rwengine/src/render/GameRenderer.hpp

```diff
@@ -140,8 +140,9 @@ class GameRenderer {
     void setupRender();
     void renderPostProcess();
 
-    Renderer* getRenderer() {
-        return renderer.get();
+    Renderer*& getRenderer() {
```

Comment:
It should just be a raw pointer if not returning as a reference otherwise this is very confusing and actually wrong as it allows the caller to change the pointer value.